src/core/utils/CorePlugin.ts

```diff
@@ -20,7 +30,96 @@ export class CorePlugin extends JWPlugin {
     // Public
     //--------------------------------------------------------------------------
 
-    removeSide(intent: Action): void {
+    removeSide(intent: Intent): void {
         console.log('REMOVE SIDE:' + intent);
     }
+    /**
+     * Navigate to a given Range (in the payload of the Intent).
+     *
+     * @param intent
+     */
+    navigate(intent: Intent): void {
+        this._setRange(intent.payload['value']);
+    }
+    /**
+     * Render the `vDocument`.
+     */
+    render(): void {
+        this.vDocument.renderer.render(this.vDocument.editable);
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a position in the `VDocument` as a tuple `[VNode, 'before' | 'after']`
+     * (ie. the position is before or after a given `VNode`). The position is
+     * always given on the leaf.
+     *
+     * @param container
+     * @param offset
+     */
+    _getPosition(container: DOMElement | Node, offset: number): VPosition {
```

Comment:
Why is a plugin playing with DOM elements ? IMO, only the parser and renderer should have references to DOM elements, all the other parts of the editor should completely ignore them and only work on our virtual document.

Reply:
As discussed, we will indeed ensure that this is done by the `Renderer`. I had done it here because at the time what we had agreed upon was to have this `CorePlugin` handle the connection between the DOM and the `VDocument`. But that was always meant to be temporary, until the memory structure is introduced and the whole world falls on its head.